catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -407,13 +413,139 @@ protected void printStatus(long count) {
+    } catch (IOException e) {
+      LOGGER.debug("Failed to add metacard with id {}.", metacard.getId(), e);
+    } finally {
+      try {
+        inputStream.close();
+      } catch (IOException | NullPointerException ignore) {
+      }
+    }
+  }
+
+  private boolean hasLocalResources(Metacard metacard) {
+    URI uri = metacard.getResourceURI();
+    return (uri != null && ContentItem.CONTENT_SCHEME.equals(uri.getScheme()));
+  }
+
+  private Map<String, Resource> getAllMetacardContent(Metacard metacard) {
+    Map<String, Resource> resourceMap = new HashMap<>();
+    Attribute attribute = metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
+
+    if (attribute != null) {
+      List<Serializable> serializables = attribute.getValues();
+      serializables.forEach(
+          serializable -> {
+            String fragment = CONTENT + File.separator;
+            URI uri = null;
+            try {
+              uri = new URI((String) serializable);
+              String derivedResourceFragment = uri.getFragment();
+              if (ContentItem.CONTENT_SCHEME.equals(uri.getScheme())
+                  && StringUtils.isNotBlank(derivedResourceFragment)) {
+                fragment += derivedResourceFragment + File.separator;
+                Resource resource = getResource(metacard);
+                if (resource != null) {
+                  resourceMap.put(
+                      fragment + uri.getSchemeSpecificPart() + "-" + resource.getName(), resource);
+                }
+              }
+            } catch (URISyntaxException e) {
+              LOGGER.debug(
+                  "Invalid Derived Resource URI Syntax for metacard : {}", metacard.getId(), e);
+            }
+          });
+    }
+
+    URI resourceUri = metacard.getResourceURI();
+
+    Resource resource = getResource(metacard);
+
+    if (resource != null) {
+      resourceMap.put(
+          CONTENT + File.separator + resourceUri.getSchemeSpecificPart() + "-" + resource.getName(),
+          resource);
+    }
+
+    return resourceMap;
+  }
+
+  private Resource getResource(Metacard metacard) {
+    Resource resource = null;
+
+    try {
+      ResourceRequest resourceRequest = new ResourceRequestById(metacard.getId());
+      ResourceResponse resourceResponse = catalogFramework.getLocalResource(resourceRequest);
+      resource = resourceResponse.getResource();
+    } catch (IOException | ResourceNotFoundException | ResourceNotSupportedException e) {
+      LOGGER.debug("Unable to retrieve content from metacard : {}", metacard.getId(), e);
+    }
+    return resource;
+  }
+
+  private void writeResourceToZip(
+      ZipOutputStream zipOutputStream, String filename, Resource resource) {
+
+    try {
+      ZipEntry zipEntry = new ZipEntry(filename);
+      zipOutputStream.putNextEntry(zipEntry);
+
+      InputStream inputStream = resource.getInputStream();
+
+      IOUtils.copy(inputStream, zipOutputStream);
+      inputStream.close();
```

Comment:
✏️ Should we put the `resource.getInputStream()` inside of a try-with-resources block so we don't have to manually close it after copying?